index.js

```diff
@@ -25,7 +25,12 @@ function compress(input, params, cb) {
     process.nextTick(cb, new Error('Second argument is not a function.'));
     return;
   }
-  encode.compressAsync(input, params, cb);
+  var stream = new TransformStreamEncode(params);
+  var chunks = [];
+  stream.on('error', cb);
+  stream.on('data', function(c) { chunks.push(c); });
+  stream.on('end', function() { cb(null, Buffer.concat(chunks)); });
```

Comment:
> I haven't benchmarked myself but I assume it'd be wise to count the length every time we push a new chunk.

I was actually the one who wrote the benchmarking for `Buffer.concat` in Node core, and there’s a small, but existing difference in performance. So yeah, makes sense.